Library/Homebrew/cmd/update.sh

```diff
@@ -696,6 +696,7 @@ EOS
      [[ -n "${HOMEBREW_UPDATE_FAILED}" ]] ||
      [[ -n "${HOMEBREW_MISSING_REMOTE_REF_DIRS}" ]] ||
      [[ -n "${HOMEBREW_UPDATE_FORCE}" ]] ||
+     [[ -n "${HOMEBREW_LINUXBREW_MIGRATION}" ]] ||
```

Comment:
This means that `update-report.rb` will be called if a migration is needed. In the case where the `version_scheme` changes were already made but the remote url gets changed back to linuxbrew for some reason, `HOMEBREW_LINUXBREW_MIGRATION` will not be set so `update-report` will not run (unless one of the other conditions is met, obviously)